src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -120,15 +120,12 @@ fn borrowck<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, owner_def_id: DefId)
         used_mut_nodes: RefCell::new(FxHashSet()),
     };
 
-    // Eventually, borrowck will always read the MIR, but at the
-    // moment we do not. So, for now, we always force MIR to be
-    // constructed for a given fn, since this may result in errors
-    // being reported and we want that to happen.
-    //
-    // Note that `mir_validated` is a "stealable" result; the
-    // thief, `optimized_mir()`, forces borrowck, so we know that
-    // is not yet stolen.
-    tcx.mir_validated(owner_def_id).borrow();
+    // Force the MIR-based borrow checker to run; this also forces the
```

Comment:
I don't 100% understand your logic in making this change part of this PR.

If you had not already added a guard elsewhere with a "if tcx.sess.nll(), then do not call bug()", then maybe I could understand forcing MIR borrowck to run.

But why do you need to force MIR borrow check to be run when you have the aforementioned guard in place on the ICE? Doesn't that avoid the problem (since we will call `bug()` when we are not in `feature(nll)` mode in that case?)

Reply:
I think you are correct that this change is not needed. I *thought* it would allow me to use span-delay-bug initially but that turned out not to be the case.